### DISCUSSION TITLE
Support for Google Spanner as an additional database in OpenFGA

### DISCUSSION THREAD

**Discussion by prakashsatyani** [openfga/community#294] [Ideas]
### Checklist

- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Describe the problem you'd like to have solved

Currently OpenFGA supports only PostgreSQL or MySQL. However for applications that want to use some high performance databases like Google Spanner or Amazon Aurora or Azure CosmosDB can not do so. It would be really helpful if we don't have the database option constraint.

### Describe the ideal solution

To add support for other databases such as,

- Google Spanner
- Amazon Aurora
- Azure CosmosDB
- MongoDB
- etc

**Comment by miparnisari** [comment]
Hi @prakashsatyani, we are discussing internally within the team how to make the addition of new supported databases a smoother and faster experience, both for the community and the maintainers. Sadly, we don't have a specific date for when the design will be finalized, but I'll post here when we have updates. Stay tuned :)

**Comment by jon-whit** [comment]
@prakashsatyani in the meantime you can use AWS RDS or AWS Aurora Postgres or MySQL as a managed offering that our existing list of supported storage engines support out of the box.

As for supporting Spanner or CosmosDB, like @miparnisari mentioned we're entertaining a pluggable datastore mechanism.

See: [Roadmap item for Pluggable Datastores](https://github.com/orgs/openfga/projects/1?pane=issue&itemId=58130782)

**Comment by aaguiarz** [comment]
Duplicated of https://github.com/orgs/openfga/discussions/179
